pkg/build/controller/build/build_controller.go

```diff
@@ -1055,6 +1055,13 @@ func (bc *BuildController) resolveImageReferences(build *buildv1.Build, update *
 		// If we cannot resolve the output reference, the output image stream
 		// may not yet exist. The build should remain in the new state and show the
 		// reason that it is still in the new state.
+
+		// However, if source type is binary, it means, that user is creating it via
+		// the 'oc start-build'. in that case it makes sense to fail the build to let
+		// the user know early that resource does not exist
+		if build.Spec.Source.Type == buildv1.BuildSourceBinary {
+			update.setPhase(buildv1.BuildPhaseFailed)
+		}
```

Comment:
Two items:

- I would use transitionToPhase here, and move this after the "No integrated registry check" below.
- Does transitioning to "failed" cause `oc start-build` to abort the source upload?